specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-08-01/frontdoor.json

```diff
@@ -0,0 +1,2484 @@
+        },
+        "x-ms-examples": {
+          "List Front Doors in a Resource Group": {
+            "$ref": "./examples/FrontdoorList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/frontDoors/{frontDoorName}": {
+      "get": {
+        "tags": [
+          "FrontDoors"
+        ],
+        "description": "Gets a Front Door with the specified Front Door name under the specified subscription and resource group.",
+        "operationId": "FrontDoors_Get",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/frontDoorNameParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/FrontDoor"
+            }
+          },
+          "default": {
+            "description": "Front Door error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Front Door": {
+            "$ref": "./examples/FrontdoorGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "FrontDoors"
+        ],
+        "description": "Creates a new Front Door with a Front Door name under the specified subscription and resource group.",
+        "operationId": "FrontDoors_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/frontDoorNameParameter"
+          },
+          {
+            "description": "Front Door properties needed to create a new Front Door.",
+            "in": "body",
+            "name": "frontDoorParameters",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/FrontDoor"
+            }
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/FrontDoor"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new Front Door has been created.",
+            "schema": {
+              "$ref": "#/definitions/FrontDoor"
+            }
+          },
+          "202": {
```

Comment:
For async, using the 201 + non-terminal provisioning state + azure async operation header is the recommended path. Looks like here you are using the 202 + location header pattern. NRP uses 202 pattern for existing types. but for new types, please consider 201. There are advantages of using the 201 pattern.